hack/build-client.sh

```diff
@@ -19,41 +19,49 @@ function checksum() {
 }
 
 function generate_golang() {
-  local API_VERSION=$1
-  local FOLDER=$2
+  local AUTOREST_IMAGE=$1
+  local API_VERSION=$2
+  local FOLDER=$3
 
   sudo docker run \
-		--rm \
-		-v $PWD/pkg/client:/github.com/Azure/ARO-RP/pkg/client:z \
-		-v $PWD/swagger:/swagger:z \
-		azuresdk/autorest \
-		--go \
-		--license-header=MICROSOFT_APACHE_NO_VERSION \
-		--namespace=redhatopenshift \
-		--input-file=/swagger/redhatopenshift/resource-manager/Microsoft.RedHatOpenShift/"$FOLDER"/"$API_VERSION"/redhatopenshift.json \
-		--output-folder=/github.com/Azure/ARO-RP/pkg/client/services/redhatopenshift/mgmt/"$API_VERSION"/redhatopenshift
+    --rm \
+    -v $PWD/pkg/client:/github.com/Azure/ARO-RP/pkg/client:z \
+    -v $PWD/swagger:/swagger:z \
+    "${AUTOREST_IMAGE}" \
+    --go \
+    --version=3.4.5 \
```

Comment:
Confused about this `--version` is that for the image of `3.3.2`?  Because it goes after the `go` in this docker run command, it's a little confusing looking at this.

Reply:
`3.3.2` is the version of the AutoRest CLI itself.
The `--version` argument is the version of @autorest/core/dist:
`--version=<string>            use the specified version of the autorest-core extension`
Those are different.